#!/usr/bin/env python
import roslib; roslib.load_manifest('rosmouse')
from dynamic_reconfigure.parameter_generator import *
from numpy import finfo

gen = ParameterGenerator()

filter_enum = gen.enum([gen.const('median_filter', int_t, 1, 'median'),
                        gen.const('kalman_filter', int_t, 2, 'kalman')],
                        'filter_type'
)

gen.add('filter_type', int_t, 0, 'Filter Type', 2, 1, 2, edit_method=filter_enum)
gen.add('window_size', int_t, 0, 'Size of the filter\'s window', default=3, min=1, max=100)
gen.add('cov_mul', double_t, 0, 'Covariance matrix multiplier', default=1.0, min=float(finfo(float).eps), max=10)


exit(gen.generate('rosmouse', 'rosmouse', 'Mouse'))
